### ISSUE TITLE
Cannot use the library behind firewall or without internet connection.

### ISSUE THREAD

**Issue by SpikingNeuron** [facultyai/dash-bootstrap-components#358]
Before opening:

- [Search for duplicate or closed issues](https://github.com/facultyai/dash-bootstrap-components/issues?utf8=%E2%9C%93&q=is%3Aissue)
- Read the [contributing guidelines](https://github.com/facultyai/dash-bootstrap-components/blob/master/.github/CONTRIBUTING.md)

Please fill out the below information as much as possible.

- dash version: `#1.11.0`
- dash-bootstrap-components version: `#0.9.2`
- components affected by bug: Cannot install on machine without internet

### What is happening?

I download dependencies on machine with internet connection using
```bat
pip download dash-bootstrap-components
```

And then install your library using below command.

```bat
pip install dash-bootstrap-components-0.9.2.tar.gz
```

I get the below error:
```txt
Processing c:\users\nxa06038\desktop\tt\dash-bootstrap-components-0.9.2.tar.gz
  Installing build dependencies ... error
  ERROR: Command errored out with exit status 1:
   command: 'c:\program files\python38\python.exe' 'c:\program files\python38\lib\site-packages\pip' install --ignore-installed --no-user --prefix 'C:\Users\nxa06038\AppData\Local\Temp\pip-build-env-5ov3teeq\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'setuptools>=40.8.0' wheel
       cwd: None
  Complete output (7 lines):
  WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x000001AB42B06460>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed')': /simple/setuptools/
  WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x000001AB42B06A30>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed')': /simple/setuptools/
  WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x000001AB42B06C40>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed')': /simple/setuptools/
  WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x000001AB42B06E50>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed')': /simple/setuptools/
  WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x000001AB42B06DC0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed')': /simple/setuptools/
  ERROR: Could not find a version that satisfies the requirement setuptools>=40.8.0 (from versions: none)
  ERROR: No matching distribution found for setuptools>=40.8.0
  ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\program files\python38\python.exe' 'c:\program files\python38\lib\site-packages\pip' install --ignore-installed --no-user --prefix 'C:\Users\nxa06038\AppData\Local\Temp\pip-build-env-5ov3teeq\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'setuptools>=40.8.0' wheel Check the logs for full command output.
```

Both updated versions for setuptools and wheel are there already installed on offline machine. Looks like you installation again looks for setuptools on internet and fails to connect.

### What should be happening?

Given that all dependencies are installed on offline machine I except that `pip install dash-bootstrap-components-0.9.2.tar.gz` common should install when internet is not available or when behind firewall.

**Comment by SpikingNeuron** [comment]
I also tried `--no-dependencies` option but that to does not work. Please fix this so that it helps users behind firewall.

**Comment by SpikingNeuron** [comment]
Just found a solution. I compile wheel from tar file using `pip wheel ...`and then used that wheel for offline installation. I assume that all that is needed from your side is to upload whl files to pypi repository :)

**Comment by tcbegley** [comment]
Thanks for raising @SpikingNeuron. The issue is that for some reason it tries to install `setuptools` as a build dependency, whether or not it's already installed, I've never really got to the bottom of that (I think that's part of the reason `--no-dependencies` doesn't work, because it doesn't ignore build dependencies).

Thanks for the suggestion of uploading wheels, I'll look into that and hopefully get this resolved :)

**Comment by tcbegley** [comment]
I just had another look into it, and I think I've found the problem thanks to [this issue](https://github.com/pypa/pip/issues/8043#issuecomment-614027307). 

Following the guidance there, I successfully managed to do the following in a virtual environment which had only `pip`, `setuptools` and `wheel` installed.

```
pip download dash-bootstrap-components
```

Then, with no internet connection

```
pip install --no-build-isolation --no-index --find-links . ./dash-bootstrap-components-0.9.2.tar.gz
```

This way `dash-bootstrap-components` and all dependencies were installed successfully.
